crates/wascc-provider/src/states/starting.rs

```diff
@@ -139,23 +148,23 @@ async fn start_container(
 
 /// The Kubelet is starting the Pod.
 #[derive(Default, Debug, TransitionTo)]
-#[transition_to(Running, Error)]
+#[transition_to(Running)]
 pub struct Starting;
 
 #[async_trait::async_trait]
-impl State<PodState> for Starting {
-    async fn next(self: Box<Self>, pod_state: &mut PodState, pod: &Pod) -> Transition<PodState> {
+impl State<ProviderState, PodState> for Starting {
+    async fn next(
+        self: Box<Self>,
+        provider_state: std::sync::Arc<tokio::sync::RwLock<ProviderState>>,
+        pod_state: &mut PodState,
+        pod: &Pod,
+    ) -> Transition<ProviderState, PodState> {
         info!("Starting containers for pod {:?}", pod.name());
 
         let mut container_handles = HashMap::new();
         for container in pod.containers() {
-            let port_assigned = match assign_container_port(
-                Arc::clone(&pod_state.shared.port_map),
-                &pod,
-                &container,
-            )
-            .await
-            {
+            let port_map = (&provider_state).read().await.s.port_map.clone();
```

Comment:
Nit: Can we change the "shared pod state" name from `s` to something more descriptive? Reading this code really tripped me up until I went back and figured out what the heck `s` was

Reply:
Oh heck did I not get rid of `s`?  I meant to merge `ProviderState` and `SharedPostState` in both providers - `ProviderState::s` was an interim measure intended to be non-invasive while I was exploring and evaluating.  Yes, this should absolutely go away.

Reply:
Okay this should now be fixed - apologies for the oversight.